src/api.js

```diff
@@ -0,0 +1,253 @@
+        'Content-type': 'application/json; charset=UTF-8',
+      },
+    });
+    return res.json();
+  } catch (error) {
+    return [];
+  }
+}
+
+export const renderMovieDetail = async (id) => {
+  const movies = await getMovies();
+  const comments = await getComments(id);
+  const projectDetailsPopupContainer = document.createElement('div');
+  projectDetailsPopupContainer.id = 'popup-container';
+
+  const projectDetailsPopupOverlay = document.createElement('div');
+  projectDetailsPopupOverlay.id = 'popup-overlay';
+  projectDetailsPopupOverlay.style.display = 'block';
+  projectDetailsPopupContainer.appendChild(projectDetailsPopupOverlay);
+
+  const moviePopup = document.createElement('div');
+  moviePopup.id = 'details-popup';
+  moviePopup.style.display = 'flex';
+  if (window.screen.width >= 768) {
+    moviePopup.style.top = `${window.pageYOffset + 77}px`;
+  } else {
+    moviePopup.style.top = `${window.pageYOffset + 16}px`;
+  }
+
+  const popupHeaderCloseButton = document.createElement('button');
+  popupHeaderCloseButton.id = 'popup-close-button';
+  popupHeaderCloseButton.innerHTML = 'Close';
+  moviePopup.appendChild(popupHeaderCloseButton);
+
+  movies.forEach((movie) => {
+    if (id === movie.show.id) {
+      const comedyDetailMovie = document.createElement('div');
+      comedyDetailMovie.classList.add('movie-div-detail');
+      const comedyDetailImage = document.createElement('img');
+      comedyDetailImage.classList.add('movie-image-detail');
+      comedyDetailImage.src = movie.show.image.original;
+
+      const comedyDetailcontainer = document.createElement('div');
+      comedyDetailcontainer.id = 'detail-container';
+
+      const movieDetailTitle = document.createElement('label');
+      movieDetailTitle.innerHTML = ` Title: ${movie.show.name}`;
+      const movieDetailStatus = document.createElement('label');
+      movieDetailStatus.innerHTML = `Status: ${movie.show.status}`;
+      const movieDetailPremiered = document.createElement('label');
+      movieDetailPremiered.innerHTML = `Premiered: ${movie.show.premiered}`;
+
+      const commentDisplay = document.createElement('h2');
+      commentDisplay.innerHTML = 'Comments';
+      const commentList = document.createElement('ul');
+      commentList.id = `${movie.show.id}`;
+      let commentCounter = 0;
+      const commnetNumber = document.createElement('span');
+
+      if (comments.length > 0) {
+        comments.forEach((comment) => {
+          commentCounter += 1;
+          commnetNumber.innerHTML = `(${commentCounter})`;
+          commentDisplay.append(commnetNumber);
+
+          const singleComment = document.createElement('li');
+          const commentUser = `${comment.username}`;
+          const commentContent = `${comment.comment}`;
+          const commentDate = `${comment.creation_date}`;
+          singleComment.innerHTML = `${commentDate} ${commentUser} : ${commentContent}`;
+          commentList.append(singleComment);
+        });
```

Comment:
- [ OPTIONAL] 🤔  What do you think to add validations error message to the comment.username and comment.comment inputs fields so as to prevent the user submitting or create an empty comment without any name and comment 👍🏻